modeshape-jcr/src/main/java/org/modeshape/jcr/JcrValueFactoryWithHint.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.jcr;
+
+import java.io.InputStream;
+
+import org.modeshape.jcr.value.BinaryValue;
+
+/**
+ * @author Wessel Nieboer
+ */
+public class JcrValueFactoryWithHint extends JcrValueFactory {
```

Comment:
Should there be explicit documentation to the effect that the configured hint is the *default* hint and won't be used for calls to `#createBinary(InputStream, String)`?

Reply:
Sounds like a good addition. Will implement

Reply:
Added javadoc.